src/components/modals/changeStatusModal/ChangeStatusModal.tsx

```diff
@@ -0,0 +1,86 @@
+import { Button, Group, Select, Stack, Textarea } from "@mantine/core";
+import { isNotEmpty, useForm } from "@mantine/form";
+import { ContextModalProps } from "@mantine/modals";
+import { useCallback, useMemo } from "react";
+import { FormErrorMessages } from "../../../lib/formErrors.ts";
+import { TUseProject } from "../../../lib/operators/types.ts";
+
+export type ChangeStatusFormValues = {
+  collaboratorId?: string;
+  notes?: string;
+};
+
+export type TChangeStatusModalInnerProps<T> = {
+  project: TUseProject;
+  defaultValues?: ChangeStatusFormValues;
+  handleSubmit: (values: T) => void;
+};
+
+export function ChangeStatusModal({
+  id,
+  context,
+  innerProps: { project, defaultValues, handleSubmit },
+}: ContextModalProps<TChangeStatusModalInnerProps<ChangeStatusFormValues>>) {
+  const form = useForm<ChangeStatusFormValues>({
+    initialValues: defaultValues ?? {
+      collaboratorId: "",
+      notes: "",
+    },
+    validate: {
+      collaboratorId: isNotEmpty(FormErrorMessages.required),
+      notes: isNotEmpty(FormErrorMessages.required),
```

Comment:
I set it as required but the type is `undefined | string`. This because during the meetup we had the PMs were asking for it to be required ("we need to know who updated it and why"). 

I was thinking about making it optional for some statuses, for example I guess that "Done" will almost never need a note.. 

Otherwise, it could be made REQUIRED for some statuses, for example `Fail` (why did it fail?), `Blocked` (what are we waiting for) or similar.. 

I surely think this has to be discuss further!

Reply:
Ok, I set it as optional!